packages/react-scripts/template/README.md

```diff
@@ -850,6 +851,48 @@ In the future we plan to integrate it into Create React App even more closely.
 
 To learn more about Flow, check out [its documentation](https://flow.org/).
 
+## Adding TypeScript
+
+TypeScript is a typed superset of JavaScript that compiles to plain JavaScript.
+
+Recent versions of [TypeScript](https://www.typescriptlang.org/) work with Create React App projects out of the box thanks to Babel 7. Beware that Babel 7 TypeScript does not allow some features of TypeScript such as constant enum and namespaces.
+
+To add TypeScript to a Create React App project, follow these steps:
+
+1. Run `npm install --dev typescript @types/react @types/react-dom @types/jest` (or `yarn add --dev typescript @types/react @types/react-dom @types/jest`).
```

Comment:
Remove the `--dev`, we have people install everything as a production dependency.

Reply:
@Timer could you explain the reasoning behind this? typescript and types are never used on production so it's a weird recommendation for the users

Reply:
At the end of the day, the application requires zero dependencies when deployed -- we generate static assets.
This means all dependencies either belong in dev or prod, but there's no point in separating them -- we chose prod dependencies so people who push and perform their builds on Heroku or Now don't have to go through the extra step of specifying to install dev dependencies.